src/main/java/org/testng/TestNG.java

```diff
@@ -1070,12 +1100,31 @@ private static void usage() {
   }
 
   private void generateReports(List<ISuite> suiteRunners) {
+
+    XMLReporterConfig config = new XMLReporterConfig();
```

Comment:
The `ReporterConfig` could be created only once at the creation of the class. With a getter (`getReporterConfig()`), it will be possible to override the default configuration.
The current reporter config methods could be deprecated at the same time.